src/app-layer-parser.c

```diff
@@ -130,7 +130,9 @@ typedef struct AppLayerParserProtoCtx_
     uint8_t first_data_dir;
 
     /* Option flags such as supporting gaps or not. */
-    uint64_t flags;
+    uint32_t option_flags;
```

Comment:
Maybe declare a cocinnelle check on flag here. This is getting messy to have 2 flags.